src/contracts/SafeEntrypoint.sol

```diff
@@ -160,14 +137,21 @@ contract SafeEntrypoint is SafeManageable, ISafeEntrypoint {
   /// @inheritdoc ISafeEntrypoint
   function unqueueAction(bytes32 _actionHash) external isAuthorized {
     // Check if the action exists
-    if (actionExecutableAt[_actionHash] == 0) revert ActionNotFound();
+    if (actions[_actionHash].executableAt == 0) revert ActionNotFound();
 
     // Check if the action has already been executed
-    if (executed[_actionHash]) revert ActionAlreadyExecuted();
+    if (actions[_actionHash].executed) revert ActionAlreadyExecuted();
+
+    // If this is a batch, unqueue all action contracts
+    if (actions[_actionHash].isBatch) {
```

Comment:
ok, we're spending the `length > 1` when queuing an action, and "saving a bit of gas" here, idk if it's needed, also you're missing the `else` statement where the only contract of the action gets unqueued, i think we should do here the `contracts > 0` and loop (and if `contracts == 0` means is an arbitrary action)